test/common/upstream/cluster_manager_impl_test.cc

```diff
@@ -1860,8 +1856,7 @@ TEST_F(ClusterManagerImplTest, MergedUpdatesOutOfWindow) {
   // The first update should be applied immediately, because even though it's mergeable
   // it's outside the default merge window of 3 seconds (found in debugger as value of
   // cluster.info()->lbConfig().update_merge_window() in ClusterManagerImpl::scheduleUpdate.
-  EXPECT_CALL(time_system_, monotonicTime())
-      .WillRepeatedly(Return(MonotonicTime(std::chrono::seconds(60))));
+  time_system_.sleep(std::chrono::seconds(60));
```

Comment:
why sleep here and below instead of setSystemTime as above if we are expanding the API. Or I guess by the same line of thinking why not use sleep above, and not expand the api?

Reply:
The point of this test and the one below it is to synthesize a delay whose value is inside or outside the merge window, and I think idiomatically that is done most concisely with a single sleep() call.

The other tests are just trying to produce a deterministic time for easy protobuf comparison, so setting an absolute time makes sense in those cases.